### PR TITLE
Moving test for input .win to pytest-regressions

### LINKED ISSUE #53: Migrate from (ab-)using 'cache' for test data to 'pytest-datadir' and / or 'pytest-regressions'
Currently, the ``compare_data`` fixture uses the caching feature of pytest to save reference data (which is generated the first time the test is run). Since this is not the intended purpose of the cache, this should be migrated to use either ``pytest-datadir`` or ``pytest-regressions``. 

See also #50.

### PR DESCRIPTION
**Pull request by normarivano**
In this way we can drop the `.pytest_cache` folder

Also, fixed the writer to create the same input in py2 and py3
(the problem was the default number of digits when converting
a float, now we are using `18.10f` everywhere).

Fixes #53

### CODE REVIEW

File: tests/test_win_writer.py [hunk 1]
```diff
@@ -1,18 +1,14 @@
...
-
 from gaas_sample import create_gaas_win_params
 
 
-def test_create_win_string(configure, create_gaas_win_params, compare_equal):
+def test_create_win_string(configure, create_gaas_win_params, file_regression):
     from aiida_wannier90.io._write_win import _create_win_string
 
-    # The 'tag' is an ugly workaround because win_writer changes
-    # behavior between Python versions 2 and 3.
```
Comment (greschd) [inline tests/test_win_writer.py]: I think we still need to figure out some way to work around the differences between Py2 and Py3 here - maybe change the extension to `.py2.win` or `.py3.win` depending on the version?

**Comment by greschd** [comment]
To make sure the pre-commit tests run, you should run `pre-commit install` in your local repository, and make sure you have the exact versions specified in the `dev` extra installed (`pip install .[dev]`).

**Comment by greschd** [comment]
Not sure what the other test failures are about - I triggered a build of `develop` manually [here](https://travis-ci.org/aiidateam/aiida-wannier90/builds/609212644?utm_medium=notification&utm_source=github_status) to check if that still works.

File: tests/conftest.py [hunk 2]
```diff
@@ -24,28 +26,3 @@
 def test_name(request):
     """Returns module_name.function_name for a given test"""
     return request.module.__name__ + '/' + request._parent_request._pyfuncitem.name
```
Comment (greschd) [inline tests/conftest.py]: This can probably also be removed.

File: tests/test_win_writer.py (hunk 1 above)
Comment (greschd) [reply tests/test_win_writer.py]: Ah sorry, I didn't see your comment about this. For some reason though, this test fails on the Python 2 travis build.

Comment (greschd) [reply tests/test_win_writer.py]: Aha, I think in Python 3 something is automatically converted to a float where it should be an int. See the difference between the deleted and current test files in the `l=...` and `mr=...` part of the projections.

File: tests/test_win_writer/test_create_win_string.win [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      0.5000000000       0.5000000000       0.0000000000
+      0.5000000000       0.5000000000       0.5000000000
+end kpoints
+
+begin projections
+random
+c=     -1.4200470448,      1.4200470448,      1.4200470448:l=     -3.0000000000,mr=      1.0000000000:::r=      1.0000000000:
+c=     -1.4200470448,      1.4200470448,      1.4200470448:l=     -3.0000000000,mr=      2.0000000000:::r=      1.0000000000:
+c=     -1.4200470448,      1.4200470448,      1.4200470448:l=     -3.0000000000,mr=      3.0000000000:::r=      1.0000000000:
+c=     -1.4200470448,      1.4200470448,      1.4200470448:l=     -3.0000000000,mr=      4.0000000000:::r=      1.0000000000:
```
Comment (greschd) [inline tests/test_win_writer/test_create_win_string.win]: Could be that Wannier90 doesn't like the spaces (or `l`, `mr` and `r` as float) that cause the tests to fail.

File: tests/test_win_writer.py (hunk 1 above)
Comment (normarivano) [reply tests/test_win_writer.py]: Indeed, we are now having a custom conversion for floats where needed, using standard formatting for the rest (integers), I'm going to push changes soon

File: tests/test_win_writer/test_create_win_string.win (hunk 3 above)
Comment (normarivano) [reply tests/test_win_writer/test_create_win_string.win]: It should be float vs. int (see comment above) - we're also removing spaces, anyway

Comment (normarivano) [reply tests/test_win_writer/test_create_win_string.win]: Indeed, it should be real vs float - we are fixing this and in any case removing spaces just to be sure

**Comment by giovannipizzi** [comment]
@greschd now the PR works - I'm going to approve it, but if you want to give a look before I merge please let me know, since I co-authored the commit

**Comment by greschd** [comment]
Very nice, thanks!

**Pull request merged**